_episodes_rmd/04-r-data-cleaning-merging.Rmd

```diff
@@ -31,151 +32,167 @@ knitr_fig_path("04-")
 
-We can resolve this warning by telling `read_csv()` what the column names should be with the `col_names()` argument where we give it the column names we want within the c() function separated by commas. If we do this, then we need to set skip to 2 to also skip the column headings. Let's also save this dataframe to `co2_emissions_dirty` so that we don't have to read it in every time we want to clean it even more.
+We previously saw how we can subset rows from a data frame using `filter()`. Let's subset out the rows we are interested in, namely those from 2019. We can also drop the `year_id` column, since we know these data are only from 2019. Let's store the resulting dataset in an object named `pollution_2019_dirty`.
 
-```{r CleanEmissionsColNames1}
-co2_emissions_dirty <- read_csv("data/co2-un-data.csv", skip=2,
-         col_names=c("region", "country", "year", "series", "value", "footnotes", "source"))
-co2_emissions_dirty
+```{r filter2019}
+pollution_2019_dirty <- ambient_pollution_dirty %>%
+  filter(year_id == 2019) %>%
+  select(-year_id)
 ```
 
-> ## Bonus: Another way to deal with this error
-> 
-> There are often multiple ways to clean data. Here we  read in the table, get the warning and then fix the column names using the rename function.
-> 
-> ```{r CleanEmissionsRename}
-> read_csv("data/co2-un-data.csv", skip=1) %>%
->   rename(country=X2)
-> ```
->
-> Many data analysts prefer to have their column headings and variable names be in all lower case. We can use a variation of `rename()`, which is `rename_all()` that allows us to set all of the column headings to lower case by giving it the name of the tolower function, which makes everything lowercase.
-> 
-> ```{r CleanEmissionsRenameLower}
-> read_csv("data/co2-un-data.csv", skip=1) %>%
->  rename_all(tolower)
-> ```
-{: .solution}
+Notice that this data object has 690 rows, as we expect from our previous data summary.
 
-We previously saw how we can subset columns from a data frame using the select function. There are a lot of columns with extraneous information in this dataset, let's subset out the columns we are interested in. 
+Sometimes it can be helpful to explore your data summaries in the view tab. Try it in the exercise below.
 
-> ## Reviewing selecting columns
-> Select the country, year, series, and value columns from our dataset. 
+> ## Viewing data summaries
+> Count the number of observations for each unique `location_name` in `pollution_2019_dirty` and use the *pipe operator* and `view()` to explore the summary data. Click on the column names to reorder the summary data however you'd like. Do you notice anything interesting?
 > 
 > > ## Solution: 
-> > ```{r CleanEmissionsSelect}
-> > 
-> > co2_emissions_dirty %>%
-> >   select(country, year, series, value)
+> > ```{r count_location_name}
+> > pollution_2019_dirty %>%
+> >   group_by(location_name) %>%
+> >   count() %>%
```

Comment:
```suggestion
> >   count() %>% # this is a shortcut for summarize(n = n())
```

Because I didn't teach them `count()` in the data analysis lesson.